src/components/button/button.spec.js

```diff
@@ -412,4 +412,42 @@ describe('Button', () => {
       }, buttonWithServiceIcon.toJSON(), { modifier: `${StyledIcon}` });
     });
   });
+
+  describe('when the fullWidth prop is provided', () => {
+    it('does not throw when size is large and buttonType is `secondary`', () => {
+      expect(() => {
+        render({
+          children: 'foo',
+          size: 'large',
+          fullWidth: true,
+          buttonType: 'secondary'
+        });
+      }).not.toThrow();
+    });
+
+    it('does not throw when size is large and buttonType is `dashed`', () => {
+      expect(() => {
+        render({
+          children: 'foo',
+          size: 'large',
+          fullWidth: true,
+          buttonType: 'dashed'
+        });
+      }).not.toThrow();
+    });
+
+    it('applies the expected style to the button', () => {
```

Comment:
an `it.each` or `describe.each` might be more effective here to iterate over the combinations and assert the relevant styling is overwritten